### PR TITLE
feat: changed snippet js declaration; added snippet not declaration

### PR DESCRIPTION
**Pull request by AlexanderBazukevich**

### CODE REVIEW

File: src/core/base/snippet.ts [hunk 1]
```diff
@@ -7,12 +7,23 @@ export type UIPSnippetTemplate = HTMLTemplateElement | HTMLScriptElement;
...
+  /** @returns snippet's js element */
   @memoize()
   public get $elementJS(): UIPSnippetTemplate | null {
     const $root: Element = this.$element.closest('uip-root') || document.body;
     const selectors = [];
-    if (this.$element.id) selectors.push(`[uip-js-snippet="${this.$element.id}"]`);
-    if (this.label) selectors.push(`[uip-js-snippet][label="${this.label}"]`);
+    if (this.jsSnippetId) selectors.push(`[id="${this.jsSnippetId}"]`);
+    if (this.label) selectors.push(`[id][label="${this.label}"]`);
+    return $root.querySelector(selectors.join(','))!;
```
Comment (ala-n) [inline src/core/base/snippet.ts:17]:
```suggestion
    return $root.getElementById(this.jsSnippetId);
```

**Review by ala-n — APPROVED** [review]

File: src/plugins/note/note.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import {listen, memoize} from '@exadel/esl/modules/esl-utils/decorators';
+import {UIPPlugin} from '../../core/base/plugin';
+
+export class UIPNote extends UIPPlugin {
+  static is = 'uip-note';
+  static title = 'Note! ';
+
+  @memoize()
+  protected get $inner(): HTMLElement {
+    return (<div class={UIPNote.is + '-inner uip-plugin-inner uip-plugin-inner-bg'}></div>) as HTMLElement;
```
Comment (dshovchko) [inline src/plugins/note/note.tsx:12]:
```suggestion
    return (<div className={`${UIPNote.is}-inner uip-plugin-inner uip-plugin-inner-bg`}></div>) as HTMLElement;
```

File: src/plugins/note/note.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'jsx-dom';
+
+import {listen, memoize} from '@exadel/esl/modules/esl-utils/decorators';
+import {UIPPlugin} from '../../core/base/plugin';
+
+export class UIPNote extends UIPPlugin {
+  static is = 'uip-note';
+  static title = 'Note! ';
```
Comment (dshovchko) [inline src/plugins/note/note.tsx:8]:
```suggestion
  public static override is = 'uip-note';
  public static title = 'Note! ';
```

**Review by dshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

File: src/plugins/note/note.tsx (hunk 3 above)
Comment (ala-n) [reply src/plugins/note/note.tsx:8]: Thanks! I'll udate it on the final preparations if not in that PR.

**Review by abarmina — APPROVED** [review]

File: src/plugins/note/note.tsx (hunk 2 above)
Comment (ala-n) [reply src/plugins/note/note.tsx:12]: According to PReact semantic (which is applicable to jsx-dom) both `class` and `className` can be used

@yadamskaya, as we have partially applied `class` and partially `className` on the project, it's up to you to finally decide which one is gonna be used then we can consider adding a rule to our ESLint Config

**Pull request merged**